### PR TITLE
feature / View finding logic for MessageViewHolder

### PR DESCRIPTION
**Pull request by bhm**
Please ensure that your pull request meets the following requirements - thanks !

* Follows our existing [codestyle](https://github.com/k9mail/k-9/wiki/CodeStyle).
* Does not break any unit tests.
* Contains a reference to the issue that it fixes.
* For cosmetic changes add one or multiple images, if possible.

### CODE REVIEW

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt [hunk 1]
```diff
@@ -400,6 +375,12 @@ class MessageListAdapter internal constructor(
     }
 }
 
+var View.visible: Boolean
```
[the remark this answers is not in the document]
Comment (bhm) [reply app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: I always forget about this existing.

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageViewHolder.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import android.widget.ImageView
+import android.widget.TextView
+import com.fsck.k9.ui.ContactBadge
+import com.fsck.k9.ui.R
+
+class MessageViewHolder(
+        view: View,
+        private val itemActionListener: MessageListItemActionListener
+) : View.OnClickListener {
+    val subject: TextView by lazy { view.findViewById(R.id.subject) }
```
[the remark this answers is not in the document]
Comment (bhm) [reply app/ui/src/main/java/com/fsck/k9/fragment/MessageViewHolder.kt]: done

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt [hunk 3]
```diff
@@ -125,50 +125,27 @@ class MessageListAdapter internal constructor(
...
 
 
         // 1 preview line is needed even if it is set to 0, because subject is part of the same text view
         holder.preview.setLines(max(appearance.previewLines, 1))
         appearance.fontSizes.setViewTextSize(holder.preview, appearance.fontSizes.messageListPreview)
-        holder.threadCount = view.findViewById(R.id.thread_count)
         appearance.fontSizes.setViewTextSize(holder.threadCount, appearance.fontSizes.messageListSubject) // thread count is next to subject
-        view.findViewById<View>(R.id.selected_checkbox_wrapper).visibility =
-                if (appearance.checkboxes) View.VISIBLE else View.GONE
+        view.findViewById<View>(R.id.selected_checkbox_wrapper).visible = appearance.checkboxes
```
[the remark this answers is not in the document]
Comment (bhm) [reply app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: done

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt [hunk 4]
```diff
@@ -220,9 +197,8 @@ class MessageListAdapter internal constructor(
             holder.flagged.isChecked = flagged
         }
         holder.position = cursor.position
-        if (holder.contactBadge != null) {
```
[the remark this answers is not in the document]
Comment (bhm) [reply app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: this is redone

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt [hunk 5]
```diff
@@ -17,35 +18,18 @@ import android.view.View
...
 import android.widget.CursorAdapter
 import android.widget.TextView
-
+import androidx.core.view.isVisible
 import com.fsck.k9.Account
-import com.fsck.k9.ui.R
 import com.fsck.k9.Preferences
 import com.fsck.k9.contacts.ContactPictureLoader
 import com.fsck.k9.controller.MessageReference
+import com.fsck.k9.fragment.MLFProjectionInfo.*
```
Comment (bhm) [inline app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: Code style from the `settings.jar` and Kotlin still defaults  to wildcard. Do we want this import style in Kotlin?

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt [hunk 6]
```diff
@@ -125,50 +125,27 @@ class MessageListAdapter internal constructor(
...
-            appearance.fontSizes.setViewTextSize(holder.from, appearance.fontSizes.messageListSender)
+        val holder = MessageViewHolder(view, listItemListener)
+        holder.contactBadge.visible = appearance.showContactPicture
 
-        } else {
-            holder.subject = view.findViewById(R.id.subject)
-            appearance.fontSizes.setViewTextSize(holder.subject, appearance.fontSizes.messageListSubject)
-
-        }
+        val subjectFontSize = if (appearance.senderAboveSubject) appearance.fontSizes.messageListSender
```
[the remark this answers is not in the document]
Comment (bhm) [reply app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: done, extracted to a val

**Comment by bhm** [comment]
It fails on downloading `kotlin-reflect` artifact for `autodiscovery` module.

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt (hunk 5 above)
Comment (cketti) [reply app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: The `settings.jar` is pretty old. I believe it doesn't contain any Kotlin-specific settings. So the IDE's defaults are used.

Comment (bhm) [reply app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: Done

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (bhm) [reply app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: Separated into commits.

**Pull request merged**